homeassistant/components/calendar/google.py

```diff
@@ -64,23 +68,46 @@ def __init__(self, calendar_service, calendar_id, search,
         self.ignore_availability = ignore_availability
         self.event = None
 
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    def update(self):
-        """Get the latest data."""
+    def _prepare_query(self):
         from httplib2 import ServerNotFoundError
 
         try:
             service = self.calendar_service.get()
         except ServerNotFoundError:
             _LOGGER.warning("Unable to connect to Google, using cached data")
             return False
-
         params = dict(DEFAULT_GOOGLE_SEARCH_PARAMS)
-        params['timeMin'] = dt.now().isoformat('T')
         params['calendarId'] = self.calendar_id
         if self.search:
             params['q'] = self.search
 
+        return service, params
+
+    async def async_get_events(self, start_date, end_date):
+        """Get all events in a specific time frame."""
+        service, params = self._prepare_query()
+        params['timeMin'] = start_date.isoformat('T')
+        params['timeMax'] = end_date.isoformat('T')
+
+        events = service.events()  # pylint: disable=no-member
+        result = events.list(**params).execute()
```

Comment:
Is this doing I/O? Because you're inside the event loop and are not allowed to do I/O

Reply:
No. Just run it inside the worker pool.

```python
result = await hass.async_add_job(events.list(**params).execute)
```